firewood/src/merkle.rs

```diff
@@ -1989,4 +1962,216 @@ mod tests {
+        );
+    }
+
+    #[test]
+    fn new_leaf_is_sibling_of_old_leaf() {
+        let key = vec![0xff];
+        let val = vec![1];
+        let key_2 = vec![0xff, 0x00];
+        let val_2 = vec![2];
+        let key_3 = vec![0xff, 0x0f];
+        let val_3 = vec![3];
+
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        merkle.insert(&key, val.clone(), root).unwrap();
+        merkle.insert(&key_2, val_2.clone(), root).unwrap();
+        merkle.insert(&key_3, val_3.clone(), root).unwrap();
+
+        assert_eq!(
+            merkle.get(&key, root).unwrap().as_deref(),
+            Some(val.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_2, root).unwrap().as_deref(),
+            Some(val_2.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_3, root).unwrap().as_deref(),
+            Some(val_3.as_slice())
+        );
+    }
+
+    #[test]
+    fn old_branch_is_a_child_of_new_branch() {
+        let key = vec![0xff, 0xf0];
+        let val = vec![1];
+        let key_2 = vec![0xff, 0xf0, 0x00];
+        let val_2 = vec![2];
+        let key_3 = vec![0xff];
+        let val_3 = vec![3];
+
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        merkle.insert(&key, val.clone(), root).unwrap();
+        merkle.insert(&key_2, val_2.clone(), root).unwrap();
+        merkle.insert(&key_3, val_3.clone(), root).unwrap();
+
+        assert_eq!(
+            merkle.get(&key, root).unwrap().as_deref(),
+            Some(val.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_2, root).unwrap().as_deref(),
+            Some(val_2.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_3, root).unwrap().as_deref(),
+            Some(val_3.as_slice())
+        );
+    }
+
+    #[test]
+    fn overlapping_branch_insert() {
+        let key = vec![0xff];
+        let val = vec![1];
+        let key_2 = vec![0xff, 0x00];
+        let val_2 = vec![2];
+
+        let overwrite = vec![3];
+
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        merkle.insert(&key, val.clone(), root).unwrap();
+        merkle.insert(&key_2, val_2.clone(), root).unwrap();
+
+        assert_eq!(
+            merkle.get(&key, root).unwrap().as_deref(),
+            Some(val.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_2, root).unwrap().as_deref(),
+            Some(val_2.as_slice())
+        );
+
+        merkle.insert(&key, overwrite.clone(), root).unwrap();
+
+        assert_eq!(
+            merkle.get(&key, root).unwrap().as_deref(),
+            Some(overwrite.as_slice())
+        );
+
+        assert_eq!(
+            merkle.get(&key_2, root).unwrap().as_deref(),
+            Some(val_2.as_slice())
+        );
+    }
+
+    #[test]
+    fn single_key_proof_with_one_node() {
```

Comment:
How is this different from `test_one_element_proof` in the tests directory?

Reply:
It's not. Would you like me to delete the other test?

Reply:
Sure. Those tests belong here anyway.